crates/datafusion-app/src/config.rs

```diff
@@ -195,6 +197,81 @@ pub struct ObjectStoreConfig {
     pub huggingface: Option<Vec<HuggingFaceConfig>>,
 }
 
+#[derive(Clone, Debug, Deserialize)]
+pub struct DatabaseConfig {
+    #[cfg(feature = "mysql")]
+    pub mysql: Option<Vec<MySQLConfig>>,
+}
+
+#[cfg(feature = "mysql")]
+#[derive(Clone, Debug, Deserialize)]
+pub struct MySQLConfig {
+    #[serde(default = "default_mysql_name")]
+    pub name: String,
+    #[serde(default = "default_mysql_host")]
+    pub host: String,
+    #[serde(default = "default_mysql_port")]
+    pub port: u16,
+    #[serde(default = "default_mysql_user")]
+    pub user: String,
+    #[serde(default = "default_mysql_password")]
+    pub password: String,
+    #[serde(default = "default_mysql_database")]
+    pub database: String,
+    #[serde(default = "default_mysql_sslmode")]
+    pub sslmode: String,
+}
+
+#[cfg(feature = "mysql")]
+fn default_mysql_name() -> String {
+    "mysql".to_string()
+}
+
+#[cfg(feature = "mysql")]
+fn default_mysql_host() -> String {
+    "127.0.0.1".to_string()
+}
+
+#[cfg(feature = "mysql")]
+fn default_mysql_port() -> u16 {
+    3306
+}
+
+#[cfg(feature = "mysql")]
+fn default_mysql_user() -> String {
+    "root".to_string()
+}
+
+#[cfg(feature = "mysql")]
+fn default_mysql_password() -> String {
+    "".to_string()
+}
+
+#[cfg(feature = "mysql")]
+fn default_mysql_database() -> String {
+    "".to_string()
+}
```

Comment:
I'm not that familiar with MySql but im surprised the default database is an empty string - is that correct?

Reply:
actually, we should remove default. MySQL does not have default database for applications and when using `mysql` client, we need to say `use {database};`